code/Player/PropPossession.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System;
+using Sandbox;
+using TTT.UI;
+
+namespace TTT;
+
+public partial class PropPossession : EntityComponent<Player>
+{
+	[Net] public Prop Prop { get; set; }
+	[Net, Local] public int Punches { get; set; }
+
+	// Clientside:
```

Comment:
```suggestion
	// Clientside
```
and move these below the server side variables imo.